### DISCUSSION TITLE
Fitting horizontal tune

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#611] [Python AT]
Hi everyone,

I am using the function fit_tune for the Fcc-ee z lattice through my orbit and optics corrections implemented in PyAT
I used fit_tune after appling shift errors to the arc quadrupoles of value sigma= 10 micro randomly distributed with cut off at 2.5sigma 
I noticed that the vertical tune is fitted well while the horizontal one is not, an example:

fit_tune(ring, get_refpts(ring, 'QF*'),
get_refpts(ring, 'QD*'),[0.2602333 , 0.37990665]  )

Fitting Tune...
Initial value [0.25203866 0.37591178]
iter# 0 Res. 0.0004360366681724843
iter# 1 Res. 0.002409322744671066
iter# 2 Res. 0.013324557952700237
Final value [0.14480133 0.37976749]

I tried to reduce the error values to 1 micron, and i see the same effect.

Thank you in advanced,

**Comment by swhite2401** [comment]
Dear @elafmusa,
It seems the fit is diverging which is strange, what is the orbit at the end of you fit? Do you see large variations?

**Comment by elafmusa** [comment]
Dear @swhite2401 ,

I see that the orbit dose not change a lot:

rms orbit  x: 9.401504462441926 mu m,  rms orbit y: 37.641367341992726 mu m

fit_tune(ring, get_refpts(ring, 'QF*'),
get_refpts(ring, 'QD*'),   [0.2602333,  0.37990665] , )

Fitting Tune...
Initial value [0.25477621 0.37561414]
iter# 0 Res. 0.00021275114446161803
iter# 1 Res. 0.0011762031626341089
iter# 2 Res. 0.006504884990431527
Final value [0.17958048 0.37981934] 

rms orbit  x: 9.38948243175231 mu m,  rms orbit y: 37.587432828408915 mu m

**Comment by swhite2401** [comment]
Ok thanks, could you send me your script so I can take a look?

**Reply by elafmusa** [reply]
Yes, sure, i will send vie email, thanks

**Comment by simoneliuzzo** [comment]
Dear @elafmusa ,

I have done such correction for FCC lattice with errors (the results are for example in A.Franchi et al. "Analytic derivative... " IPAC23-MOPL069).
 
I am using a smaller set of quadrupoles:

focussing_quadrupoles_for_tune = 'QF[0-9]*'
defocussing_quadrupoles_for_tune = 'QD[0-9]*'

and the option to match integer part of the tunes as well

at.fit_tune returns

Fitting Tune...
Initial value [214.25914351 214.39671412]
iter# 0 Res. 2.9676679308821273e-08
iter# 1 Res. 2.5070366228645863e-12
iter# 2 Res. 2.1195460261586413e-16
Final value [214.26023811 214.37990139]

Probably my initial errors are smaller then yours as well.

best regards
Simone

**Reply by elafmusa** [reply]
Dear @simoneliuzzo and @swhite2401,
Thank you for the information!, using the integer part of the tunes have solved my problem. Though i was able before to fit the fractional part using different lattice.
